apps/auth/xmpp/auth.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2014 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+from apps.auth.service import AuthService
+from superdesk import get_resource_service
+from superdesk.resource import Resource
+from apps.auth.errors import UserDisabledError, CredentialsAuthError
+from superdesk.errors import SuperdeskApiError
+from superdesk.users.errors import UserInactiveError
+from superdesk import utils
+from flask import current_app as app
+import requests
+import superdesk
+
+
+class XMPPAuthResource(Resource):
+    schema = {
+        'jid': {
+            'type': 'string',
+            'required': True
+        },
+        'transactionId': {
+            'type': 'string',
+            'required': True
+        },
+        'token': {
+            'type': 'string'
+        },
+        'user': Resource.rel('users', True)
+    }
+    resource_methods = ['POST']
+    item_methods = ['GET', 'DELETE']
```

Comment:
imo there should be no item methods, only `POST` to create new tokens, tokens are fetched/removed via `auth` resource right?